scrape/manager.go

```diff
@@ -110,6 +110,7 @@ func (m *Manager) reload() {
 			scrapeConfig, ok := m.scrapeConfigs[setName]
 			if !ok {
 				level.Error(m.logger).Log("msg", "error reloading target set", "err", "invalid config id:"+setName)
+				m.mtxScrape.Unlock()
```

Comment:
Thanks for catching it! A `defer m.mtxScrape.Unlock()` following L104 would be easier to read and maintain IMO.

Reply:
This method has a `sync.WaitGroup` to wait some goroutines finishing. So it should release `m.mtxScrape` as soon as possible.

Reply:
Hmm, in fact the problem is that we shouldn't return in the middle of the loop but rather continue to the next item. @krasi-georgiev?

Reply:
In this case, we only can check if `m.targetSets` and `m.scrapeConfigs` has same keys. But it still does not guarantee that their configs are consistent.

Reply:
IIUC the scrape manager configuration has been reloaded but the discovery manager hasn't been yet. So it could send target updates for scraping configurations that don't exist anymore. It is just a transient discrepancy between the 2 managers and I would assume that it is safe to skip the group as the scrape manager knows the latest configuration.

Reply:
yes replacing `return` with `continue` would be most appropriate here.